Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,222 @@
+    fold_test_scores <- NULL
+    for(i in 1 : n_folds){
+      prediction <- predict(fold_tree[[i]], feature.mat[fold_split_idx$test[i,],])
+      fold_test_scores <- c(fold_test_scores, scorer(target.mat, prediction))
+    }
+      
+    best_alpha <-  0.
+    best_score <- mean(fold_test_scores)
+    best_tree <- master_tree
+    alphas <- c(0.)
+    alpha_cv_scores <- c(best_score)
+    
+    ### calc train score
+    prediction <- predict(master_tree, feature.mat)
+    alpha_train_scores <- scorer(target.mat, prediction)
+    
+    ### calc cost of all leaves
+    if(learner == "mmif"){
+      for(i in 1:length(master_tree)){
+        ter_id <- nodeids(master_tree[[i]], terminal = TRUE)
+        n <- nodeapply(master_tree[[i]], ids = ter_id, info_node)
+        alpha_train_objective_values <- alpha_train_objective_values + sum(matrix(unlist(n), nrow = length(n), byrow = T)[, 2])
+      }
+      alpha_train_objective_values <- alpha_train_objective_values/length(master_tree)
+    }
+    else{
+      ter_id <- nodeids(master_tree, terminal = TRUE)
+      n <- nodeapply(master_tree, ids = ter_id, info_node)
+      alpha_train_objective_values <- sum(matrix(unlist(n), nrow = length(n), byrow = T)[, 2])
+    }
+    
+
+  }
+  
+  ### Generate a big dictionnary of all HP combinations considered (including alpha) and their CV scores
+  ### Append alpha to the parameters
+  best_params <- parameters
+  
+  if(learner == "mmit"){
+    best_params$alpha <- best_alpha
+    cv_results <- cbind(parameters$max_depth, parameters$margin, parameters$min_sample, parameters$loss,
+                        alphas, alpha_cv_scores, alpha_train_scores, alpha_train_objective_values)
+    colnames(cv_results) <- c("max_depth", "margin", "min_sample", "loss", "alpha", " cv_score", "train_score", "train_objective_value")
+    cv_results <- as.data.frame(cv_results)
+  }
+  else{
+    cv_results <- cbind(parameters$max_depth, parameters$margin, parameters$min_sample, parameters$loss,
+                        parameters$n_trees, parameters$n_features, alpha_train_scores, alpha_train_objective_values)
+    colnames(cv_results) <- c("max_depth", "margin", "min_sample", "loss", "n_trees", " n_features", "train_score", "train_objective_value")
+    cv_results <- as.data.frame(cv_results)
+  }
+  
+  output <- list()
+  output$best_score <- best_score
+  output$best_estimator <- best_tree
+  output$best_params <- best_params
+  output$cv_results <- cv_results
+  
+  return(output)
+  
+}, ex=function(){
+  
+  data(neuroblastomaProcessed, package="penaltyLearning")
+  feature.mat <- data.frame(neuroblastomaProcessed$feature.mat)[1:45,]
+  target.mat <- neuroblastomaProcessed$target.mat[1:45,]
+
+  parameters <- list()
+  parameters$max_depth <- Inf
+  parameters$margin <- 2
+  parameters$min_sample <- 2
+  parameters$loss <- c("hinge")
+  
+  result <- fit_and_score(target.mat, feature.mat, parameters, learner = "mmit", predict = mmit.predict, scorer = mse, pruning = FALSE)
```

Comment:
When you change the arguments to remove predict, dont forget to adapt the examples

Reply:
done